packages/lib/src/interactions/SelectToZoom.tsx

```diff
@@ -1,69 +1,76 @@
 import { useThree } from '@react-three/fiber';
 
 import { useVisCanvasContext } from '../vis/shared/VisCanvasProvider';
-import RatioSelectionRect from './RatioSelectionRect';
 import SelectionRect from './SelectionRect';
 import SelectionTool from './SelectionTool';
-import { useMoveCameraTo } from './hooks';
-import type { CommonInteractionProps, Selection } from './models';
-import { getEnclosedRectangle, getRatioRespectingRectangle } from './utils';
+import Box from './box';
+import { useZoomOnBox } from './hooks';
+import type { CommonInteractionProps, Rect3, Selection } from './models';
 
 type Props = CommonInteractionProps;
 
 function SelectToZoom(props: Props) {
-  const { canvasSize, canvasRatio, visRatio, dataToWorld } =
+  const { canvasSize, canvasRatio, visRatio, worldToData } =
     useVisCanvasContext();
 
   const { width, height } = canvasSize;
   const keepRatio = visRatio !== undefined;
 
   const camera = useThree((state) => state.camera);
-  const moveCameraTo = useMoveCameraTo();
+  const zoomOnBox = useZoomOnBox();
 
-  function onSelectionEnd(selection: Selection) {
-    const [worldStart, worldEnd] = getRatioRespectingRectangle(
-      ...selection.data,
-      keepRatio ? canvasRatio : undefined
-    ).map(dataToWorld); // work in world coordinates as we need to act on the world camera
+  function computeZoomBox(worldSelection: Rect3): Box {
```

Comment:
This function is called from both the render prop, to compute the zoom box to display via `SelectionRect`, and from `onSelectionEnd`, to compute the box to pass to `zoomOnBox`.

First, it creates a box for the given `worldSelection`, then, if `keepRatio` is enabled, it expands the box to match the ratio and then "keeps" this box within the camera's FOV. To compute the FOV box, I use the same trick as discussed for `moveCameraTo`: I multiply the scale by the canvas size.

As explained, this "fixes" the behaviour of `SelectToZoom` slightly, as the zoom box used to be kept within the visualization area, not within the FOV.